packages/user-ui/src/pages/project/components/sample/sample.rs

```diff
@@ -46,7 +46,7 @@ pub fn get_survey_status(started_at: i64, ended_at: i64) -> SurveyStatus {
 
 #[component]
 pub fn Sample(lang: Language) -> Element {
-    let mut ctrl = controller::SampleController::init(lang)?;
+    let mut ctrl: Controller = use_context();
```

Comment:
It may causes inconsistent data modification.
If you can, you should return data by EventHandler, and Controller sholud be handled only by Page component .

Sometimes, if you must use controller directly by architectural issue, you should only use it to limited parts.
But now, your code is using controller for convinience. It seems to be a little anti-pattern.

In the aspect of Service, they are designed to be shared by other components.
It does mean that they mostly contain shared object.
But Controller is usually designed for contraint scope.
So if you call controller outside of Page component, you must consider scope of all data.
But it is hard to handle scopes of all data, namely violate SRP.

Reply:
@hackartists 
The part that concerns me is that if basic information, sample survey, deliberation, discussion, voting, etc. are managed in one controller and passed as parameters, the controller code will become longer and the parameters will actually increase. Would it be better to manage them in the main and pass them as parameters?

Or would it be right to create a separate controller for each of the parts and implement it by initializing each controller?